src/quo2/components/avatars/channel_avatar/view.cljs

```diff
@@ -0,0 +1,63 @@
+(ns quo2.components.avatars.channel-avatar.view
+  (:require [clojure.string :as string]
+            [quo2.components.avatars.channel-avatar.style :as style]
+            [quo2.components.icon :as icons]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            utils.string))
+
+(defn- initials
+  [full-name amount-initials color]
+  [text/text
+   {:accessibility-label :initials
+    :size                :paragraph-2
+    :number-of-lines     1
+    :ellipsize-mode      :clip
+    :weight              :semi-bold
+    :style               {:color color}}
+   (utils.string/get-initials full-name amount-initials)])
+
+(defn- lock
+  [locked? big?]
+  ;; When `locked?` is nil, we must not display the unlocked icon.
+  (when (boolean? locked?)
+    [rn/view
+     {:accessibility-label :lock
+      :style               (style/lock-container {:big? big?})}
+     [icons/icon (if locked? :i/locked :i/unlocked)
+      {:color           (colors/theme-colors colors/neutral-50 colors/neutral-40)
+       :container-style style/lock-icon
+       :size            12}]]))
+
+(defn view
+  "Options:
+
+  :big? - bool (default nil) - Container size
+
+  :emoji - string (default nil)
+
+  :color - color (default nil) If the component is used for a community channel,
+  then the default color should be the community custom color.
+
+  :locked? - nil/bool (default nil) - When true/false display a locked/unlocked
+  icon respectively. When nil does not show icon.
+
+  :full-name - string (default nil) - When :emoji is blank, this value will be
+  used to extract the initials.
+
+  :amount-initials - int (default 1) - Number of initials to be extracted
+  from :full-name when :emoji is blank.
```

Comment:
I'm skeptical about adding an `int` option to control the number of initial letters.

From my understanding, there are only 2 valid options 1 letter or 2 letters. When I looked at the design what I understood is that when the channel avatar is big we should display 2 letters and when it is small 1 letter.
<img width="238" alt="Screenshot 2023-06-24 at 12 52 59" src="https://github.com/status-im/status-mobile/assets/19279756/d15802f5-64c1-47f6-8c2a-43eea9c383d4">

Also, when displaying two letters on a small avatar the second letter is being cut off in some cases.
<img width="300" src="https://github.com/status-im/status-mobile/assets/19279756/c594a931-c83d-43f8-925c-7b85703574b7"/>

Reply:
there's `size` -> `amount-initials` logic in `user-avatar` https://github.com/status-im/status-mobile/pull/16379/files#diff-89879ff7bdd5d29334dbca769f65209502d91884ecbd9436f99fbd9f4946931bR11
don't know if this applies to `channel-avatar`

Reply:
The final code is cleaner now after I removed the `amount-initials` parameter, thanks again.